### PR TITLE
MITRE Duplicates 

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://github.com/demisto/etc/issues/38463

## Description
Added technique prefix to all sub technique, 
Added suffix `[Malware]` to duplicates malware with intrusion sets.

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 1]
```diff
@@ -162,11 +167,44 @@ def build_iterator(self, create_relationships=False, is_up_to_6_2=True, limit: i
...
+            mitre_id_to_mitre_name[mitre_id[0]] = mitre_item_json.get('name')
+
+
+def add_technique_prefix_to_sub_technique(indicators, id_to_name, mitre_id_to_mitre_name):
+    for indicator in indicators:
+        if indicator['type'] in ['Attack Pattern', 'STIX Attack Pattern'] and \
+                len(indicator['fields']['mitreid']) > 5:  # Txxxx.xxx is sub technique
+            parent_mitre_id = indicator['fields']['mitreid'][:5]
+            value = indicator['value']
+            technique = mitre_id_to_mitre_name.get(parent_mitre_id)  # TODO in case that technique doesnt exist
```
Comment (yuvalbenshalom) [inline Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:192]: Finish TODO

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 2]
```diff
@@ -162,11 +167,44 @@ def build_iterator(self, create_relationships=False, is_up_to_6_2=True, limit: i
...
+            new_value = f'{technique}: {value}'
+            indicator['value'] = new_value
+            id_to_name[indicator['fields']['stixid']] = new_value
+
+
+def add_malware_prefix_to_dup_with_intrusion_set(indicators):
```
Comment (yuvalbenshalom) [inline Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:198]: maybe it worth explaining what this function is for.
also, note that this pattern of functions that are manipulating the arguments should generally be avoided.
lets talk about it.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comments

File: Packs/FeedMitreAttackv2/ReleaseNotes/1_0_2.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### MITRE ATT&CK Feed v2
+- **Breaking changes**: Added to the sub-technique (Attack Pattern) prefix of the parent technique.
+- **Breaking changes**: Added (Malware) suffix to the **Malware** objects in case that there is **Intrusion Set** with same name.
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Packs/FeedMitreAttackv2/ReleaseNotes/1_0_2.md:5]: done

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 4]
```diff
@@ -117,11 +117,16 @@ def build_iterator(self, create_relationships=False, is_up_to_6_2=True, limit: i
...
         mitre_relationships_list = []
         id_to_name: Dict = {}
+        mitre_id_to_mitre_name: Dict = {}
         counter = 0
 
         # For each collection
         for collection in self.collections:
 
+            # fetch only enterprise objects
+            if collection.id != '95ecc380-afe9-11e4-9b6c-751b66dd541e':
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:127]: done

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py (hunk 2 above)
Comment (ChanochShayner) [reply Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:198]: done

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py (hunk 1 above)
Comment (ChanochShayner) [reply Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:192]: There are no such cases.

**Review by yuvalbenshalom — APPROVED** [review]

File: Packs/FeedMitreAttackv2/ReleaseNotes/1_0_2.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+##### MITRE ATT&CK Feed v2
+- **Breaking changes**: Added to the sub-technique (Attack Pattern) prefix of the parent technique.
```
Comment (amshamah419) [inline Packs/FeedMitreAttackv2/ReleaseNotes/1_0_2.md:4]:
```suggestion
- **Breaking changes**: Added to the sub-technique, Attack Pattern, the prefix of the parent technique.
```

File: Packs/FeedMitreAttackv2/ReleaseNotes/1_0_2.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (amshamah419) [reply Packs/FeedMitreAttackv2/ReleaseNotes/1_0_2.md:5]:
```suggestion
- **Breaking changes**: Added the Malware suffix to the **Malware** objects in cases where there is an **Intrusion Set** with the same name.
```

**Pull request merged**